config/ax_blas.m4

```diff
@@ -164,7 +164,37 @@ fi
 
 # BLAS in Intel MKL library?
 if test $ax_blas_ok = no; then
-  AC_CHECK_LIB(mkl, $sgemm, [ax_blas_ok=yes;BLAS_LIBS="-lmkl"])
+  # MKL for gfortran
+  if test x"$ac_cv_fc_compiler_gnu" = xyes; then
+    # 64 bit
+    if test $host_cpu = x86_64; then
+      AC_CHECK_LIB(mkl_gf_lp64, $sgemm,
+      [ax_blas_ok=yes;BLAS_LIBS="-lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lpthread"],,
+      [-lmkl_gf_lp64 -lmkl_sequential -lmkl_core -lpthread])
+    # 32 bit
+    elif test $host_cpu = i686; then
+      AC_CHECK_LIB(mkl_gf, $sgemm,
+        [ax_blas_ok=yes;BLAS_LIBS="-lmkl_gf -lmkl_sequential -lmkl_core -lpthread"],,
+        [-lmkl_gf -lmkl_sequential -lmkl_core -lpthread])
+    fi
+  # MKL for other compilers (Intel, PGI, ...?)
+  else
+    # 64-bit
+    if test $host_cpu = x86_64; then
+      AC_CHECK_LIB(mkl_intel_lp64, $sgemm,
+        [ax_blas_ok=yes;BLAS_LIBS="-lmkl_intel_lp64 -lmkl_sequential -lmkl_core -lpthread"],,
```

Comment:
@paullric It checks for these libraries and not `-lmkl` with Intel compilers. Did you do `autoreconf -fi` after you merged this branch ?